pkg/health/server/prober.go

```diff
@@ -346,39 +412,67 @@ func (p *prober) runHTTPProbe() {
 				})
 			}
 
-			resp := p.httpProbe(name, ip.String())
-			if resp.Status != "" {
-				scopedLog.Debugf("Failed to probe: %s", resp.Status)
+			ctx := context.Background()
+			httpWg.Add(1)
+			go p.httpProbe(name, ip.String(), ctx, httpResChan, &httpWg)
+			icmpWg.Add(1)
+			go p.icmpPing(name, ip.String(), ctx, icmpResChan, &icmpWg)
```

Comment:
I would probably simplify this a bit and move rate-limiting here. In case we have thousands of nodes, there is no point in starting thousands of goroutines, which would be waiting on rate limiter. So I would do something like this instead:
```
	if err := p.rateLimiter.Wait(ctx); err != nil {
		// Handle error
	} else {
               wg.Add(2)
               go p.httpProbe(name, ip.String(), ctx, httpResChan, &wg)
               go p.icmpPing(name, ip.String(), ctx, icmpResChan, &wg)
        }
```
And then:
```
	go func() {
		wg.Wait()
		close(icmpResChan)
                close(httpResChan)
	}()
```
Also, note that you would need to start goroutine with receive loops `for resp := range httpResChan {` and `for resp := range icmpResChan {` before starting `httpProbe` and `icmpPing` goroutines.